vulnerability-analyzer/src/main/java/org/hyades/persistence/VulnerableSoftwareRepository.java

```diff
@@ -22,20 +22,23 @@ public List<VulnerableSoftware> getAllVulnerableSoftware(final String part, fina
                             .and("purlType", purl.getType())
                             .and("purlNamespace", purl.getNamespace())
                             .and("purlName", purl.getName()));
-        }
-        else if (cpeSpecified) {
+        } else if (cpeSpecified) {
             return list("part = :part and vendor = :vendor and product = :product",
                     Parameters.with("part", part)
                             .and("vendor", vendor)
                             .and("product", product));
-        }
-        else if (purl != null) {
-            return list("purlType = :purlType and purlNamespace = :purlNamespace and purlName = :purlName",
-                    Parameters.with("purlType", purl.getType())
-                            .and("purlNamespace", purl.getNamespace())
-                            .and("purlName", purl.getName()));
-        }
-        else {
+        } else if (purl != null) {
+            if (purl.getNamespace() == null) {
+                return list("purlType = :purlType and purlName = :purlName",
+                        Parameters.with("purlType", purl.getType())
+                                .and("purlName", purl.getName()));
```

Comment:
We still need to check that the namespace is `null` here, just turns out that JPA requires special handling of `null` checks: https://stackoverflow.com/a/28562010

```suggestion
                return list("purlType = :purlType and purlNamespace is null and purlName = :purlName",
                        Parameters.with("purlType", purl.getType())
                                .and("purlName", purl.getName()));
```